core/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+pub mod basic_config;
+pub mod configuration;
+pub mod projects;
+
+extern crate anyhow;
+extern crate serde;
+extern crate shellexpand;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::anyhow;
+use basic_config::BasicConfigContent;
+use std::fs;
+use std::path::Path;
+use std::vec::Vec;
+
+pub fn setup(path: &Path, additional_modules: Vec<Box<dyn SetupAdapter>>) -> Result<()> {
+    if path.is_dir() {
+        return Err(anyhow!(
+            "This command needs a single file, that is not a directory!"
+        ));
+    }
+
+    let config_content: String = fs::read_to_string(path)
+        .with_context(|| format!("Cannot read file at path {}", path.display()))?;
+
+    let Some(extension) = path.extension().and_then(|ext| ext.to_str()) else {
+        return Err(anyhow!("Could not read extension "));
+    };
```

Comment:
```suggestion
    let extension = path.extension()
        .and_then(|ext| ext.to_str())
        .ok_or(anyhow!("Could not read extension "))?;
    };
```

nit i think this reads better but its preference

Reply:
Valuable nit, especially considering my state of knowledge at this point! Thanks!